tests/mir-opt/building/logical_and_in_conditional.rs

```diff
@@ -0,0 +1,42 @@
+struct Droppy;
+
+impl Drop for Droppy {
+    fn drop(&mut self) {
+        println!("drop");
+    }
+}
+
+// EMIT_MIR logical_and_in_conditional.test_and.built.after.mir
+fn test_and(a: i32, b: i32, c: i32) {
+    if {
+        let _t = Droppy;
+        a == 0
+    } && {
+        let _t = Droppy;
+        b == 0
+    } && {
+        let _t = Droppy;
+        c == 0
+    } {}
+}
```

Comment:
```suggestion
struct Droppy(bool);

impl Drop for Droppy {
    fn drop(&mut self) {
        println!("drop");
    }
}

// EMIT_MIR logical_and_in_conditional.test_and.built.after.mir
fn test_and(a: i32, b: i32, c: i32) {
    if Droppy(a).0 && Droppy(b).0 && Droppy(c).0 {}
}
```

Thanks. Nit: curly braces create their own scope. We want to ensure that they are dropped in the implicit scope of the condition.